plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -903,6 +903,7 @@ private RemoteTableName constructPageSinkIdsTable(ConnectorSession session, Conn
         LongWriteFunction pageSinkIdWriter = (LongWriteFunction) toWriteMapping(session, TRINO_PAGE_SINK_ID_COLUMN_TYPE).getWriteFunction();
 
         execute(session, connection, pageSinkTableSql);
+        closer.register(() -> dropTable(session, pageSinkTable, true));
```

Comment:
this `dropTable` executes `DROP TABLE` which will also fail if we don't end up creating the table in the first place.

Reply:
but the line above just created successfully that table, right?

Reply:
yes, I did not notice that this line is part of `constructPageSinkIdsTable` method and not `finishInsertTable` - downsides of doing quick reviews in browser.

Reply:
> this `dropTable` executes `DROP TABLE` which will also fail if we don't end up creating the table in the first place.

If `execute(session, connection, pageSinkTableSql); ` is failed, it will jump to the `catch` and `finally` code block in `finishInsertTable()`, which means the `closer.register(() -> dropTable(session, pageSinkTable, true)); ` will not be executed.

So if the table creation is not successful, the closer registration for dropping this table is not needed